File: iconex/src/main/java/foundation/icon/iconex/view/ui/create/CreateWalletViewModel.java
package foundation.icon.iconex.view.ui.create;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import foundation.icon.MyConstants;
import foundation.icon.iconex.wallet.Wallet;

public class CreateWalletViewModel extends ViewModel {

    private MutableLiveData<MyConstants.Coin> coinType;
    private MutableLiveData<String> privateKey;
    private MutableLiveData<Wallet> wallet;

    public MutableLiveData<MyConstants.Coin> getCoin() {
        if (coinType == null)
            coinType = new MutableLiveData<>();

        return coinType;
    }

    public void setCoinType(MyConstants.Coin coin) {
        getCoin().setValue(coin);
    }

    public MutableLiveData<String> getPrivateKey() {
        if (privateKey == null)
            privateKey = new MutableLiveData<>();

        return privateKey;
    }

    public void setPrivateKey(String privateKey) {
        getPrivateKey().setValue(privateKey);
    }

    public MutableLiveData<Wallet> getWallet() {
        if (wallet == null)
            wallet = new MutableLiveData<>();

        return wallet;
    }

    public void setWallet(Wallet wallet) {
        getWallet().setValue(wallet);
    }
}

File: iconex/src/main/java/foundation/icon/iconex/view/ui/create/CreateWalletViewModel.kt
package foundation.icon.iconex.view.ui.create

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import foundation.icon.iconex.wallet.Wallet

class CreateWalletViewModel : ViewModel() {

    private var coinType: MutableLiveData<CoinType>? = null
    private var wallet: MutableLiveData<Wallet>? = null
    private var privateKey: MutableLiveData<String>? = null

    fun getCoinType(): MutableLiveData<CoinType> {
        if (coinType == null)
            coinType = MutableLiveData()

        return coinType!!
    }

    fun setCoinType(type: CoinType) {
        getCoinType().value = type
    }

    fun getWallet(): MutableLiveData<Wallet> {
        if (wallet == null)
            wallet = MutableLiveData()

        return wallet!!
    }

    fun setWallet(wallet: Wallet) {
        getWallet().value = wallet
    }

    fun getPrivateKey(): MutableLiveData<String> {
        if (privateKey == null)
            privateKey = MutableLiveData()

        return privateKey!!
    }

    fun setPrivateKey(privateKey: String) {
        getPrivateKey().value = privateKey
    }

    enum class CoinType constructor(val type: String) {
        ICX("icx"),
        ETH("eth");


        companion object {

            fun fromType(type: String): CoinType? {
                for (t in values()) {
                    if (t.type == type)
                        return t
                }

                return null
            }
        }
    }
}
